src/coreclr/vm/virtualcallstub.cpp

```diff
@@ -95,21 +98,72 @@ extern size_t g_dispatch_cache_chain_success_counter;
 
 SPTR_IMPL_INIT(VirtualCallStubManagerManager, VirtualCallStubManagerManager, g_pManager, NULL);
 
+#ifdef FEATURE_CACHED_INTERFACE_DISPATCH
+struct CachedIndirectionCellBlockListNode
+{
+    CachedIndirectionCellBlockListNode *m_pNext;
+    TADDR m_pFiller; // Used to ensure that the Indirection Cells are double pointer aligned
+    InterfaceDispatchCell m_rgIndCells[0];
+};
+#endif // FEATURE_CACHED_INTERFACE_DISPATCH
+
 #ifndef DACCESS_COMPILE
 
+BYTE* GenerateDispatchStubCellEntryMethodDesc(LoaderAllocator *pLoaderAllocator, TypeHandle ownerType, MethodDesc *pMD, LCGMethodResolver *pResolver)
+{
+    return GenerateDispatchStubCellEntrySlot(pLoaderAllocator, ownerType, pMD->GetSlot(), pResolver);
+}
+
+BYTE* GenerateDispatchStubCellEntrySlot(LoaderAllocator *pLoaderAllocator, TypeHandle ownerType, int methodSlot, LCGMethodResolver *pResolver)
+{
+    // Generate a dispatch stub and gather a slot.
+    //
+    // We generate an indirection so we don't have to write to the dictionary
+    // when we do updates, and to simplify stub indirect callsites.  Stubs stored in
+    // dictionaries use "RegisterIndirect" stub calling, e.g. "call [eax]",
+    // i.e. here the register "eax" would contain the value fetched from the dictionary,
+    // which in turn points to the stub indirection which holds the value the current stub
+    // address itself. If we just used "call eax" then we wouldn't know which stub indirection
+    // to update.  If we really wanted to avoid the extra indirection we could return the _address_ of the
+    // dictionary entry to the  caller, still using "call [eax]", and then the
+    // stub dispatch mechanism can update the dictitonary itself and we don't
+    // need an indirection.
+
+    VirtualCallStubManager * pMgr = pLoaderAllocator->GetVirtualCallStubManager();
+
+    // We indirect through a cell so that updates can take place atomically.
+    // The call stub and the indirection cell have the same lifetime as the dictionary itself, i.e.
+    // are allocated in the domain of the dicitonary.
+    DispatchToken token = VirtualCallStubManager::GetTokenFromFromOwnerAndSlot(ownerType, methodSlot);
```

Comment:
A nit:
```suggestion
    DispatchToken token = VirtualCallStubManager::GetTokenFromOwnerAndSlot(ownerType, methodSlot);
```

Reply:
How did I not notice THAT?! Fixed in my local copy.